dll/win32/shell32/COpenWithMenu.cpp

```diff
@@ -31,6 +31,56 @@ WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
 EXTERN_C BOOL PathIsExeW(LPCWSTR lpszPath);
 
+static SIZE_T PathGetAppFromCommandLine(LPCWSTR pszIn, LPWSTR pszOut, SIZE_T cchMax)
```

Comment:
Isn't there already a suitable helper to do that? (for example from the ShellExecute implementation)
Or even, [CommandLineToArgvW()](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw).